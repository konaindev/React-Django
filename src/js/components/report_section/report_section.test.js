import renderer from "react-test-renderer";
import ReportSection from "./index";
import ReportDateSpan from "../report_date_span";

describe("ReportSection", () => {
  it("renders correctly", () => {
    const props = {
      name: "Test Name"
    };
    const tree = renderer.create(<ReportSection {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with report date span correctly", () => {
    const props = {
        name: "Test Name",
        sectionItems: (
          <ReportDateSpan
            name="Report Span"
            dates={{ start: "2018-01-01", end: "2018-02-01" }}
          />
        )
    };
    const tree = renderer.create(<ReportSection {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
