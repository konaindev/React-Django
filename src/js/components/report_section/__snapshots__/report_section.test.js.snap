// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportSection renders correctly 1`] = `
<div
  className="report-section"
>
  <div
    className="section-header"
  >
    <p
      className="section-header__title"
    >
      Test Name
    </p>
  </div>
</div>
`;

exports[`ReportSection renders with report date span correctly 1`] = `
<div
  className="report-section"
>
  <div
    className="section-header"
  >
    <p
      className="section-header__title"
    >
      Test Name
    </p>
    <div
      className="section-header__extra"
    >
      <div
        className="report-date-span__content"
      >
        <span
          className="report-date-span__content-name"
        >
          Report Span
           (
          4 wks.
          )
        </span>
        <span
          className="report-date-span__content-dates"
        >
          1/1/18
           -
           
          2/1/18
        </span>
      </div>
    </div>
  </div>
</div>
`;
